src/openvic-simulation/ThreadPool.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <condition_variable>
+#include <cstdint>
+#include <mutex>
+#include <thread>
+#include <vector>
+
+#include "openvic-simulation/pop/PopValuesFromProvince.hpp"
+#include "openvic-simulation/types/Date.hpp"
+
+namespace OpenVic {
+	struct GoodInstance;
+	struct PopsDefines;
+	struct Strata;
+
+	class ThreadPool {	
+	private:
+		enum WORK_TYPE : uint8_t {
+			NONE,
+			GOOD_EXECUTE_ORDERS,
+			PROVINCE_INITIALISE_FOR_NEW_GAME,
+			PROVINCE_TICK
+		};
+
+		void loop_until_cancelled(
+			WORK_TYPE& work_type,
```

Comment:
Don't use references for enums, this is needlessly error prone, it just causes headaches.